lib/matplotlib/tests/test_collections.py

```diff
@@ -408,6 +408,49 @@ def test_EllipseCollection():
     ax.autoscale_view()
 
 
+def test_EllipseCollection_setter_getter():
+    # Test widths, heights and angle setter
+    rng = np.random.default_rng(0)
+
+    widths = (2, )
+    heights = (3, )
+    angles = (45, )
+    offsets = rng.random((10, 2)) * 10
+
+    fig, ax = plt.subplots()
+
+    ec = mcollections.EllipseCollection(
+        widths=widths,
+        heights=heights,
+        angles=angles,
+        offsets=offsets,
+        units='x',
+        offset_transform=ax.transData,
+        )
+
+    assert_array_almost_equal(ec._widths, np.array(widths).ravel() * 0.5)
+    assert_array_almost_equal(ec._heights, np.array(heights).ravel() * 0.5)
+    assert_array_almost_equal(ec._angles, np.deg2rad(angles).ravel())
```

Comment:
Do we need to check internals now that we have the getters?

Reply:
These check that the internal values are correct (multiplication of 0.5) and these are different from the values returned by the getter, so I would say that this is still worth checking these values?

Reply:
Testing internals is not ideal, but OTOH these tests do not impose a big liability. It's unlikely that we will refactor the internals and if so, the tests can be easily adapted.

Alternatives would be an image comparison (not favored) or checking the bounding box of an EllipseCollection with a single entry.

Reply:
If this is find to keep to check as it is I would prefer that option because it is straighforward and more simple than the bounding box alternative.